Client/src/renderer/views/command-centre/CommandCentre.vue

```diff
@@ -205,31 +219,56 @@ watch(commandId, () => {
 const buttonConfig = ref<ButtonConfig[]>([
     {
         text: "Send",
-        callback: async (): Promise<void> => {
-            const validated = validateInputs(commandParamValueInputs.value);
-
-            if (!validated || !isClickable.value) {
-                return;
-            }
-
-            commandDelay();
-
-            await sendCommand();
-
-            clearInputs([commandSelectionInput.value]);
-        },
+        callback: sendCommand,
         disabled: (): boolean => !isClickable.value,
     },
     {
         text: "Leave Tyranny",
-        callback: (): void => {
-            exitTyrantView();
-        },
+        callback: exitTyrantView,
         disabled: () => false,
     },
 ]);
-</script>
 
+const submitSearch = (event: InputEvent) => {
+    const target = (event.target as HTMLInputElement);
+    const value = target.value;
+
+    if (room.value.commands.has(value)) {
+        selectedCommand.value = value;
+        target.value = '';
+        return;
+    }
+
+    const options = [...room.value.commands.values()]
+        .filter((option) => option.name.toLowerCase().includes(value.toLowerCase()))
+
+    if (options.length > 0) {
+        selectedCommand.value = options[0].name;
+        target.value = '';
+        return;
+    }
+
+    target.value = '';
+    selectedCommand.value = '';
+}
+
+window.addEventListener('keyup', (event: KeyboardEvent) => {
+    // console.log(event.ctrlKey, event.altKey, event.shiftKey, event.metaKey, event.key);
+
+    if (! commandSelectionInput.value || ! isClickable)
```

Comment:
The spaces are intentional. If you don't like it, I can change it, but recently started adding a space in between to make it a lot more clear that it is there. It's easy to overlook. Just a personal preference thing

Reply:
I do prefer not having the space because it is a unary operator like `-something` or `++something`, adding a space makes it feel like a binary operator so it looks weird 11

Reply:
Whilst that is true for sure, I'd argue you should not see it as such. It's only used in if statements and it completely inverses the result of the statement. In my opinion, specifically for the `!` it's nice to be able to see it more clearly in a statement.

I can see how `-` can also inverse a statement, but that's only for numbers, and is also very common as negative numbers are ofc very 'normal'. This can be for absolutely anything and is optional.

Reply:
I've changed them back to not have a space for consistency in the repository. This is something we can discuss at a later date in a DM 11